benchmark/cublas/blas2/spr2.cpp

```diff
@@ -0,0 +1,186 @@
+/**************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename spr2.cpp
+ *
+ **************************************************************************/
+
+#include "../utils.hpp"
+
+template <typename scalar_t>
+std::string get_name(char uplo, int size, scalar_t alpha, int incX) {
+  std::ostringstream str{};
+  str << "BM_Spr2<" << blas_benchmark::utils::get_type_name<scalar_t>() << ">/"
+      << uplo << "/" << size << "/" << alpha << "/" << incX;
+  return str.str();
+}
+
+template <typename scalar_t, typename... args_t>
+static inline void cublas_routine(args_t&&... args) {
+  if constexpr (std::is_same_v<scalar_t, float>) {
+    CUBLAS_CHECK(cublasSspr2(std::forward<args_t>(args)...));
+  } else if constexpr (std::is_same_v<scalar_t, double>) {
+    CUBLAS_CHECK(cublasDspr2(std::forward<args_t>(args)...));
+  }
+  return;
+}
+
+template <typename scalar_t>
+void run(benchmark::State& state, cublasHandle_t* cuda_handle_ptr, char uplo,
+         int size, scalar_t alpha, int incX, bool* success) {
+  // The counters are double. We convert size to double to avoid
+  // integer overflows for n_fl_ops and bytes_processed
+  double n_d = static_cast<double>(size * (size + 1) / 2);
+
+  // Temporarely set incY to 1 for all benchmarks
+  index_t incY = 1;
+
+  state.counters["n_d"] = n_d;
+  state.counters["alpha"] = static_cast<double>(alpha);
+  state.counters["incX"] = incX;
+  state.counters["incY"] = incY;
+
+  const double nflops_XtimesX = 2.0 * n_d;
+  const double nflops_tot = size + nflops_XtimesX;
+  state.counters["n_fl_ops"] = nflops_tot;
```

Comment:
There are additional operations & memory accesses in `spr2` compared to `spr`. Can you double check them and update these ? I've checked the theoretical values document and they seem correct.

Reply:
You are absolutely right! Thank you for spotting it